builder/dockerfile/internals.go

```diff
@@ -255,6 +255,7 @@ func (b *Builder) download(srcURL string) (fi builder.FileInfo, err error) {
 	// ignoring error because the file was already opened successfully
 	tmpFileSt, err := tmpFile.Stat()
 	if err != nil {
+		tmpFile.Close()
```

Comment:
Remove all of these explicit calls (unless there's a good reason for them) with a `defer tmpFile.Close()` after [this line](https://github.com/docker/docker/pull/23951/files#diff-f157a3a45b3e5d85aadff73bff1f5a7cR244).